delay login till the upload to enable save for later for guests (bug 397...

jdlrobson opened the pull request:
...11)

This proved to be a little more difficult then expected.... and I'm not sure if this is complete and worth merging or worth leaving till later.
- if username is not known then username is set as GUEST
- updated uploads to look for GUEST or the current username
- when going from welcome page to uploads page do login, if going from elsewhere
  do not (e.g. after doing save for later)

TODO: any photos saved for later will show the username as GUEST
- i guess we need to update the description text at time of upload?

jrobson committed:
delay login till the upload to enable save for later for guests (bug 39711)

* if username is not known then username is set as GUEST
* updated uploads to look for GUEST or the current username
* when going from welcome page to uploads page do login, if going from elsewhere
do not (e.g. after doing save for later)

TODO: any photos saved for later will show the username as GUEST
- i guess we need to update the description text at time of upload?

bvibber commented:
Probably the thing to do is to fill out the description template at upload time, yeah.

jrobson committed:
refactor so that photo description is made at time of upload

jrobson committed:
allow route to uploads page from welcome screen without login

show all uploads regardless of username in the uploads screen

move localise to the top so page links can be defined in messages

bvibber commented:
"By clicking the ... button below, you, GUEST, irrevocably grant ...."

Hmm, that looks kinda bad. I'd consider taking the username out of this message...

bvibber commented:
"Missing uploads? Fear no more! Simply login to see the complete list."

Couldn't we just show all uploads on the phone instead of listing by username? This might be simpler.

awjrichards-zz commented:
Per Brion - perhaps we can not display the username in the license agreement if username == GUEST, but display it if the user is logged in.

bvibber commented:
After logging in, the photo I saved for later while not logged in no longer appears in my incompletes list -- so I've got no obvious way to go back and upload it anymore....

jdlrobson commented:
I think when @yuvipanda originally did this he was allowing for different people on the same device to use the app. I guess this is unlikely so we could remove the username bit.

License text is pretty serious though... I will fix.

jrobson committed:
remove username from confirmation text

jdlrobson commented:
waiting on legal to confirm bc3b771 is ok

jdlrobson commented on assets/www/js/app.js:
```diff
@@ -163,6 +164,7 @@ require( [ 'jquery', 'l10n', 'geo', 'api', 'templates', 'monuments', 'monument',
 		var $page = $("#" + pageName); 
 		$('.page, .popup-container-container').hide(); // hide existing popups
 		$page.show()[ 0 ].scrollIntoView();
+		// if warnings exist for guests show them
```
remove me

jrobson committed:
don't alter the photo object

when you update the text of the photo object it becomes different
from the existing photo in the database. thus it never gets moved from
incomplete to completed.
